fault-monitor/operational-status-monitor.cpp

```diff
@@ -19,6 +21,49 @@ namespace status
 {
 namespace monitor
 {
+void Monitor::removeCriticalAssociation(const std::string& objectPath)
+{
+    try
+    {
+        utils::PropertyValue getAssociationValue =
+            utils::DBusHandler().getProperty(
+                objectPath.c_str(),
+                "xyz.openbmc_project.Association.Definitions", "Associations");
+
+        auto association = std::get<AssociationsProperty>(getAssociationValue);
+
+        AssociationTuple critAssociation{
+            "health_rollup", "critical",
+            "/xyz/openbmc_project/inventory/system/chassis"};
+
+        auto it =
+            std::find(association.begin(), association.end(), critAssociation);
+
+        if (it != association.end())
+        {
+            association.erase(it);
+
+            utils::PropertyValue setAssociationValue = association;
+
+            utils::DBusHandler().setProperty(
+                objectPath.c_str(),
+                "xyz.openbmc_project.Association.Definitions", "Associations",
+                setAssociationValue);
+
+            std::cout << "\nRemoved critical association between " << objectPath
+                      << " and /xyz/openbmc_project/inventory/system/chassis."
+                      << std::endl;
+        }
+    }
+    catch (const sdbusplus::exception::exception& e)
```

Comment:
do we need to catch each exception or can we use catch(...) for all of them ?

Reply:
yes as the error statement differs. I've replaced cerr with lg2::error here .